pkg/cmd/roachtest/tests/lease.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func registerLease(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:    "acquire-lease/crash/non-system",
+		Owner:   registry.OwnerKV,
+		Timeout: time.Hour,
+		Cluster: r.MakeClusterSpec(7, spec.CPU(4)),
+		Run:     runAcquireLeaseCrashNonSystem,
+	})
+}
+
+// runAcquireLeaseCrashNonSystem benchmarks the maximum duration of range
+// unavailability following a leaseholder crash. It tests the simplest possible
+// failure:
+//
+//   - A process crash, where the host/OS remains available (in particular, the
+//     TCP/IP stack is responsive and sends immediate RST packets to peers).
+//
+//   - No system ranges located on the crashed node.
+//
+//   - SQL clients do not connect to the crashed node.
+//
+//   - The workload consists of individual point reads and writes.
+//
+// We expect the worst-case (pMax) latency to be slightly larger than the lease
+// interval (9s), to account for lease acquisition and retry latencies. We do
+// not assert this, but instead export latency histograms for graphing.
+//
+// The cluster layout is as follows:
+//
+// n1-n3: System ranges and SQL gateways.
+// n4-n6: Workload ranges.
+// n7:    Workload runner.
+//
+// The test runs a kv50 workload with batch size 1, using 256 concurrent
+// workers directed at n1-n3. n4-n6 are killed and restarted in order, with 30
+// seconds between each operation, for 3 cycles totaling 9 crashes.
+func runAcquireLeaseCrashNonSystem(ctx context.Context, t test.Test, c cluster.Cluster) {
+	c.Put(ctx, t.Cockroach(), "./cockroach")
+	c.Put(ctx, t.DeprecatedWorkload(), "./workload")
+
+	// Create cluster with one rack locality per node and configure it.
+	require.Equal(t, 7, c.Spec().NodeCount)
+
+	opts := option.DefaultStartOpts()
+	settings := install.MakeClusterSettings(install.NumRacksOption(c.Spec().NodeCount))
```

Comment:
`roachprod` clusters come with some attributes included, so you shouldn't need to bother with this.

https://github.com/cockroachdb/cockroach/blob/ae0ad5ca29ff865324cfc52bf75c75357f8af592/pkg/roachprod/install/cockroach.go#L581

Reply:
Nice, thanks. Updated.